monitor/monitor.go

```diff
@@ -22,11 +31,24 @@ type monitorStats struct {
 	sthAge       *prometheus.GaugeVec
 	sthFailures  *prometheus.CounterVec
 	sthLatency   *prometheus.HistogramVec
+
+	certSubmitLatency   *prometheus.HistogramVec
```

Comment:
The two chunks of stats here made me realize we effectively have two different components here: The sth-fetching component, and the cert-submitting component. If we split those into two different structs (and two different Go files) it would probably make things cleaner. We could create both from `New()` and even compose them both into the `Monitor` struct. What do you think?

Reply:
Sure. I think that makes sense. I suspected there would be additional checks added soon and we would want to do this but I'm happy to do it as part of this PR ahead of any further checks.

Reply:
If you want to land this PR, then do the refactoring as a follow-up, I'm fine with that.